packages/itwin/tree-widget/src/components/trees/CategoriesVisibilityUtils.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import type { IModelConnection, ViewManager, Viewport } from "@itwin/core-frontend";
+import { CategoryVisibilityHandler, loadCategoriesFromViewport } from "./category-tree/CategoryVisibilityHandler";
```

Comment:
I think functions that are used by `toggleAllCategories` should be defined here and not imported from `CategoryVisibilityHandler`. 

Also maybe it would be possible to decouple `toggleAllCategories` function from `IPresentationTreeDataProvider`? Maybe instead of taking `filteredProvider` it could take an optional list of `categoryIds`. In that case category ids lookup from `filteredProvider` can be moved to `CategoriesTreeComponent`. That way this function would be more reusable between visibility trees as it would not have `CategoryTree` specific implementation details.